site/profile/manifests/users.pp

```diff
@@ -128,19 +128,38 @@
   String $selinux_user = 'unconfined_u',
   String $mls_range = 's0-s0:c0.c1023',
   String $authenticationmethods = '',
+  Boolean $manage_home = true,
+  Boolean $purge_ssh_keys = true,
+  Optional[String] $shell = '/bin/bash',
+  Optional[Integer] $uid = undef,
+  Optional[Integer] $gid = undef,
+  Optional[String] $group = $name,
```

Comment:
> It is normally a mistake to set the default of an Optional parameter to something other than undef. This is because it's not possible to 'pass' undef as the value to use for a parameter when declaring a class or defined type. When you try to set a parameter to undef, Puppet actually uses the class's default value for that parameter, not undef itself.
https://github.com/voxpupuli/puppet-lint-optional_default-check

Reply:
Mmmm, so the proper way would be 
```
  Optional[String] $group = undef,
``` 

followed by some in-class code such as 
```
$group = pick($group, $name) 
``` 
?

Reply:
Simply remove the optional. It will default to `$name`, but it can never be `undef`.

Reply:
done